Support MariaDB 10.11

phracek opened the pull request:
Support MariaDB 10.11 for building, testing, and pushing to Quay.io.

The pull request is separated to more commits.
1) git move 10.5 -> 10.11
2) copy sources back to 10.11->10.5
3) Make changes in 10.11 directory to support 10.11
4) Add build an push to quay.io

<!---

Please review the Contribution Guidelines[1] before submitting the Pull Request.

For more information about the Software Collection Organization, please visit the Welcome pages[2].

[1] https://github.com/sclorg/welcome/blob/master/contribution.md
[2] https://github.com/sclorg/welcome

-->

phracek commented:
[test]

phracek commented:
[test]

phracek commented:
[test]

zmiklank commented:
[test]

phracek commented:
[test]

zmiklank commented:
f40 mariadb10.11 is already in bodhi: https://bodhi.fedoraproject.org/updates/FEDORA-2024-fd6c145765

zmiklank commented:
It will no longer be modular, so this PR would need to be adjusted accordingly, when the rpms are out.

phracek commented:
[test]

phracek commented:
> f40 mariadb10.11 is already in bodhi: https://bodhi.fedoraproject.org/updates/FEDORA-2024-fd6c145765

Because of mariadb10.11 is not yet in branch f40. Do not test mariadb10.11 on Fedora.

phracek commented:
[test]

phracek commented:
RHEL8 and RHEL9 tests are failing:

```
==============================================
Tests were run for image rhel8/mariadb-1011:1
==============================================
Test cases results:

 [PASSED] for 'appnamenotset' run_container_creation_tests (00:00:10)
 [PASSED] for 'appnamenotset' run_configuration_tests (00:00:17)
 [PASSED] for 'appnamenotset' run_general_tests (00:01:01)
 [PASSED] for 'appnamenotset' run_change_password_test (00:00:13)
 [PASSED] for 'appnamenotset' run_change_password_new_user_test (00:00:12)
 [PASSED] for 'appnamenotset' run_replication_test (00:00:11)
 [PASSED] for 'appnamenotset' run_doc_test (00:00:01)
 [PASSED] for 'appnamenotset' run_s2i_test (00:00:23)
 [PASSED] for 'appnamenotset' run_ssl_test (00:00:08)
 [FAILED] for 'appnamenotset' run_upgrade_test (00:00:19)

```

Failed test is here:
```
-----------------------------------------------
Running test run_upgrade_test (starting at 2024-02-12 09:18:36-05:00) ... 
-----------------------------------------------
  Testing upgrade of the container image
    Created container 79b33a69f92114191ea8c1c5ed1c5ca6345c1664c61c3696d08b729b51a8a54a
  Testing MySQL connection to 10.88.0.131...
    Trying to connect...... OK
  Success!
  Testing upgrade from data without version
    Created container 4b5a8d31c2307e538609319a76a34c2d9fbcea2b5a65a4db1f295df0bc81c506
  Testing MySQL connection to 10.88.0.135...
    Trying to connect..... OK
  Success!
  Testing upgrade from too old data
  Success!
  Testing upgrade from previous version
    Created container 7a79a49ccf0f714c45668d7f3bc8f6c5646307c09afffed531392c3f73432073
  Testing MySQL connection to 10.88.0.139...
    Trying to connect..... OK
  Success!
Upgrade did not happen but it should when upgrading from previous version
=> sourcing 20-validate-variables.sh ...
=> sourcing 25-validate-replication-variables.sh ...
=> sourcing 30-base-config.sh ...
---> 14:18:50     Processing basic MySQL configuration files ...
=> sourcing 60-replication-config.sh ...
=> sourcing 70-s2i-config.sh ...
---> 14:18:50     Processing additional arbitrary  MySQL configuration provided by s2i ...
=> sourcing 40-paas.cnf ...
=> sourcing 50-my-tuning.cnf ...
---> 14:18:50     Starting MySQL server with disabled networking ...
---> 14:18:50     Waiting for MySQL to start ...
2024-02-12 14:18:50 0 [Note] Starting MariaDB 10.11.6-MariaDB source revision fecd78b83785d5ae96f2c6ff340375be803cd299 as process 26
2024-02-12 14:18:51 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2024-02-12 14:18:51 0 [Note] InnoDB: Number of transaction pools: 1
2024-02-12 14:18:51 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-02-12 14:18:51 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
2024-02-12 14:18:51 0 [Note] InnoDB: Using Linux native AIO
2024-02-12 14:18:51 0 [Note] InnoDB: Initializing buffer pool, total size = 5.000MiB, chunk size = 1.000MiB
2024-02-12 14:18:51 0 [Note] InnoDB: Completed initialization of buffer pool
2024-02-12 14:18:51 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
2024-02-12 14:18:51 0 [Note] InnoDB: End of log at LSN=46980
2024-02-12 14:18:51 0 [Note] InnoDB: 128 rollback segments are active.
2024-02-12 14:18:51 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
2024-02-12 14:18:51 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-02-12 14:18:51 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2024-02-12 14:18:51 0 [Note] InnoDB: log sequence number 46980; transaction id 14
2024-02-12 14:18:51 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-02-12 14:18:51 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/data/ib_buffer_pool
2024-02-12 14:18:51 0 [Warning] 'user' entry 'root@79b33a69f921' ignored in --skip-name-resolve mode.
2024-02-12 14:18:51 0 [Warning] 'proxies_priv' entry '@% root@79b33a69f921' ignored in --skip-name-resolve mode.
2024-02-12 14:18:51 0 [Note] /usr/libexec/mysqld: ready for connections.
Version: '10.11.6-MariaDB'  socket: '/tmp/mysql.sock'  port: 0  MariaDB Server
2024-02-12 14:18:51 0 [Note] InnoDB: Buffer pool(s) load completed at 240212 14:18:51
---> 14:18:51     MySQL started successfully
=> sourcing 40-datadir-action.sh ...
---> 14:18:52     Running datadir action: upgrade-auto
---> 14:18:52     Warning: Version of the data could not be determined. It is because the file mysql_upgrade_info is missing in the data directory, which is most probably because it was not created when initialization of data directory. In order to allow seamless updates to the next higher version in the future, the file mysql_upgrade_info will be created. If the data directory was created with a different version than 10.11, it is required to run this container with the MYSQL_DATADIR_ACTION environment variable set to 'force', or run 'mysql_upgrade' utility manually; the mysql_upgrade tool checks the tables and creates such a file as well. For upstream documentation about upgrading, see: Non expected version '10.11'
File /var/lib/mysql/data/mysql_upgrade_info exists, nothing is done.
=> sourcing 50-passwd-change.sh ...
---> 14:18:52     Setting passwords ...
2024-02-12 14:18:52 6 [Warning] 'user' entry 'root@79b33a69f921' ignored in --skip-name-resolve mode.
2024-02-12 14:18:52 6 [Warning] 'proxies_priv' entry '@% root@79b33a69f921' ignored in --skip-name-resolve mode.
---> 14:18:52     Shutting down MySQL ...
2024-02-12 14:18:52 7 [Warning] 'user' entry 'root@79b33a69f921' ignored in --skip-name-resolve mode.
2024-02-12 14:18:52 7 [Warning] 'proxies_priv' entry '@% root@79b33a69f921' ignored in --skip-name-resolve mode.
2024-02-12 14:18:52 0 [Note] /usr/libexec/mysqld (initiated by: root[root] @ localhost []): Normal shutdown
2024-02-12 14:18:52 0 [Note] InnoDB: FTS optimize thread exiting.
2024-02-12 14:18:52 0 [Note] InnoDB: Starting shutdown...
2024-02-12 14:18:52 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/data/ib_buffer_pool
2024-02-12 14:18:52 0 [Note] InnoDB: Restricted to 78 pages due to innodb_buf_pool_dump_pct=25
2024-02-12 14:18:52 0 [Note] InnoDB: Buffer pool(s) dump completed at 240212 14:18:52
2024-02-12 14:18:52 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
2024-02-12 14:18:52 0 [Note] InnoDB: Shutdown completed; log sequence number 46980; transaction id 15
2024-02-12 14:18:52 0 [Note] /usr/libexec/mysqld: Shutdown complete

---> 14:18:53     Cleaning up environment variables MYSQL_USER, MYSQL_PASSWORD, MYSQL_DATABASE and MYSQL_ROOT_PASSWORD ...
---> 14:18:53     Running final exec -- Only MySQL server logs after this point
2024-02-12 14:18:53 0 [Note] Starting MariaDB 10.11.6-MariaDB source revision fecd78b83785d5ae96f2c6ff340375be803cd299 as process 1
2024-02-12 14:18:53 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2024-02-12 14:18:53 0 [Note] InnoDB: Number of transaction pools: 1
2024-02-12 14:18:53 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-02-12 14:18:53 0 [Note] mysqld: O_TMPFILE is not supported on /var/tmp (disabling future attempts)
2024-02-12 14:18:53 0 [Note] InnoDB: Using Linux native AIO
2024-02-12 14:18:53 0 [Note] InnoDB: Initializing buffer pool, total size = 5.000MiB, chunk size = 1.000MiB
2024-02-12 14:18:53 0 [Note] InnoDB: Completed initialization of buffer pool
2024-02-12 14:18:53 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
2024-02-12 14:18:53 0 [Note] InnoDB: End of log at LSN=46980
2024-02-12 14:18:53 0 [Note] InnoDB: 128 rollback segments are active.
2024-02-12 14:18:53 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-02-12 14:18:53 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2024-02-12 14:18:53 0 [Note] InnoDB: log sequence number 46980; transaction id 14
2024-02-12 14:18:53 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/data/ib_buffer_pool
2024-02-12 14:18:53 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-02-12 14:18:53 0 [Note] Server socket created on IP: '0.0.0.0'.
2024-02-12 14:18:53 0 [Note] Server socket created on IP: '::'.
2024-02-12 14:18:53 0 [Warning] 'user' entry 'root@79b33a69f921' ignored in --skip-name-resolve mode.
2024-02-12 14:18:53 0 [Warning] 'proxies_priv' entry '@% root@79b33a69f921' ignored in --skip-name-resolve mode.
2024-02-12 14:18:53 0 [Note] /usr/libexec/mysqld: ready for connections.
Version: '10.11.6-MariaDB'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  MariaDB Server
2024-02-12 14:18:53 0 [Note] InnoDB: Buffer pool(s) load completed at 240212 14:18:53
2024-02-12 14:18:54 0 [Note] /usr/libexec/mysqld (initiated by: unknown): Normal shutdown
2024-02-12 14:18:54 0 [Note] InnoDB: FTS optimize thread exiting.
2024-02-12 14:18:54 0 [Note] InnoDB: Starting shutdown...
2024-02-12 14:18:54 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/data/ib_buffer_pool
2024-02-12 14:18:54 0 [Note] InnoDB: Restricted to 78 pages due to innodb_buf_pool_dump_pct=25
2024-02-12 14:18:54 0 [Note] InnoDB: Buffer pool(s) dump completed at 240212 14:18:54
2024-02-12 14:18:54 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
2024-02-12 14:18:54 0 [Note] InnoDB: Shutdown completed; log sequence number 46980; transaction id 15
2024-02-12 14:18:54 0 [Note] /usr/libexec/mysqld: Shutdown complete

Test for image 'rhel8/mariadb-1011:1' FAILED (exit code: 1)
Tests finished with EXIT=0

```

fila43 commented on 10.11/Dockerfile:
```diff
@@ -0,0 +1,79 @@
+#  * $MYSQL_CHARSET (Optional) - Default character set
+#  * $MYSQL_COLLATION (Optional) - Default collation
+
+ENV MYSQL_VERSION=10.5 \
```
Why should we have version 10.5 in 10.11 directory? Even Dockerfile for RHEL7 doesn't make sense to me if we do not support 10.11 in RHEL7 images

phracek replied:
Fixed by https://github.com/sclorg/mariadb-container/pull/223/commits/746dc057b928981308fd521d70ddf3ae7ece29f1

fila43 commented on 10.11/Dockerfile.c9s:
```diff
@@ -0,0 +1,70 @@
+# safe in the future. This should *never* change, the last test is there
+# to make sure of that.
+RUN INSTALL_PKGS="policycoreutils rsync tar gettext hostname bind-utils groff-base mariadb-server" && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
```
This will install the default version of mariadb i.e. 10.5. Please add module enablement for 10.11

phracek replied:
Added by commit https://github.com/sclorg/mariadb-container/pull/223/commits/746dc057b928981308fd521d70ddf3ae7ece29f1

fila43 commented on 10.11/Dockerfile.rhel7:
```diff
@@ -0,0 +1,80 @@
+# to make sure of that.
+RUN yum install -y yum-utils && \
+    prepare-yum-repositories rhel-server-rhscl-7-rpms && \
+    INSTALL_PKGS="rsync tar gettext hostname bind-utils groff-base rh-mariadb1011 rh-mariadb1011-mariadb-server-utils rh-mariadb1011-mariadb-syspaths" && \
```
I am not mariadb and container expert so maybe I am mistaken, but AFAIK rhscl does not support mariadb10.11. At lease I haven't found such branch in dist-git

phracek replied:
Fixed https://github.com/sclorg/mariadb-container/pull/223/commits/746dc057b928981308fd521d70ddf3ae7ece29f1. But there is an exclude-rhel7 file that means, that the version is not build neither tested.

zmiklank replied:
Why are we even adding the Dockerfile if there is no need for it?

phracek replied:
Fixed by https://github.com/sclorg/mariadb-container/pull/223/commits/8603271a2f27714811a83a9d68f832210c85d880

fila43 commented on 10.11/Dockerfile.rhel9:
```diff
@@ -0,0 +1,70 @@
+# safe in the future. This should *never* change, the last test is there
+# to make sure of that.
+RUN INSTALL_PKGS="policycoreutils rsync tar gettext hostname bind-utils groff-base mariadb-server" && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
```
same as the c9s you need to enable correct module

phracek replied:
Fixed by https://github.com/sclorg/mariadb-container/pull/223/commits/746dc057b928981308fd521d70ddf3ae7ece29f1

phracek commented:
[test]

hhorak commented:
There is a missing entry for 10.11 at https://github.com/sclorg/mariadb-container/blob/master/root-common/usr/share/container-scripts/mysql/init/40-datadir-action.sh#L8 -- it should likely include links to all versions between 10.5 and 10.11

hhorak commented:
I've pushed an additional commit that should help fixing the upgrade.

hhorak commented:
[test]

zmiklank commented on 10.11/root/usr/share/container-scripts/mysql/README.md:
```diff
@@ -0,0 +1,369 @@
+---------------
+This image can be extended in Openshift using the `Source` build strategy or via the standalone
+[source-to-image](https://github.com/openshift/source-to-image) application (where available).
+For this, we will assume that you are using the `rhscl/mariadb-1011-rhel7` image,
```
Also rhel7 will not get mariadb 10.11.

phracek replied:
Fixed by https://github.com/sclorg/mariadb-container/pull/223/commits/0d6ea1b236e5a3f0a9c0d28e300ecc4feb49c66e

zmiklank commented on 10.11/root/usr/share/container-scripts/mysql/README.md:
```diff
@@ -0,0 +1,369 @@
+Dockerfile and other sources for this container image are available on
+https://github.com/sclorg/mariadb-container.
+In that repository, the Dockerfile for CentOS is called Dockerfile, the Dockerfile
+for RHEL7 is called Dockerfile.rhel7, the Dockerfile for RHEL8 is called Dockerfile.rhel8,
```
Same here.

phracek replied:
Fixed by https://github.com/sclorg/mariadb-container/pull/223/commits/0d6ea1b236e5a3f0a9c0d28e300ecc4feb49c66e

zmiklank commented on 10.11/root/usr/share/container-scripts/mysql/README.md:
```diff
@@ -0,0 +1,369 @@
+MariaDB 10.11 SQL Database Server Docker image
```
This will be imo problem for more versions and images, but we should not use phrase "Docker image" and use "Container image" instead.
```suggestion
MariaDB 10.11 SQL Database Server Container image
```

phracek commented on 10.11/root/usr/share/container-scripts/mysql/README.md:
```diff
@@ -0,0 +1,369 @@
+This container image includes MariaDB 10.11 SQL database server for OpenShift and general usage.
+Users can choose between RHEL, CentOS and Fedora based images.
+The RHEL images are available in the [Red Hat Container Catalog](https://access.redhat.com/containers/),
+the CentOS images are available on [Quay.io/centos7](https://quay.io/organization/centos7),
```
Fixed by https://github.com/sclorg/mariadb-container/pull/223/commits/0d6ea1b236e5a3f0a9c0d28e300ecc4feb49c66e

phracek commented:
[test]

zmiklank commented:
Could you please squash the fix up commits before merging?

phracek commented:
Rebased
[test]

hhorak commented:
The failure of 10.5 on RHEL-9 was likely caused by a wrong compose (mariadb 10.5 was missing), it seems to be fixed now, so re-running the tests.
[test]

phracek commented:
[test]

phracek commented:
[test-openshift]

phracek commented:
[test]

zmiklank commented on imagestreams/mariadb-centos.json:
```diff
@@ -44,6 +44,42 @@
         }
       },
+      {
+        "name": "10.11-c9s",
```
name should be "10.11-el9"

phracek replied:
As you can see here: https://quay.io/organization/sclorg all our container that are part of CentOS Stream has the name either 'c9s' or 'c8s'

phracek replied:
And we do not want to rename all of them.

zmiklank replied:
Actually we were discussing this topic recently on team Slack. I was asking why also CentOS Stream imagestream tags are named -el9/-el8. Reason is, that "el" does not mean "rhel" and this is on purpose - to make the switch between CentOS Stream and RHEL imagestreams easier - and should not be changed.

Not really sure how the c8s/c9s tags got into mariadb container, but e.g., postgresql uses -el tags, as it should have. See [here](https://github.com/sclorg/postgresql-container/blob/master/imagestreams/postgresql-centos.json)

@hhorak do I understand this correctly please?

zmiklank commented on imagestreams/mariadb-centos.json:
```diff
@@ -44,6 +44,42 @@
+        }
+      },
+      {
+        "name": "10.11-c8s",
```
name should be "10.11-el8"

phracek replied:
As you can see here: https://quay.io/organization/sclorg all our container that are part of CentOS Stream has the name either 'c9s' or 'c8s'
And we do not want to rename all of them.

phracek commented:
[test]

zmiklank reviewed (approved):
LGTM, thanks.
